### PR TITLE
DLS-9710 - Added missing preservingMdc block to mongo calls

### PR DESCRIPTION
**Pull request by TomRafferty**
This PR currently stands to have jenkins run automatically upon changes pushed.

### CODE REVIEW

**Comment by TomRafferty** [comment]
> * There is a bobby warning (builds will be prevented after 31 May 2024) - see the pr builder for details

play to be updated in future tickets

**Review by karolina-zaborska — DISMISSED** [review]

File: app/uk/gov/hmrc/helptosaveapi/repo/EligibilityStore.scala [hunk 1]
```diff
@@ -95,21 +97,27 @@ class MongoEligibilityStore @Inject() (mongoComponent: MongoComponent, servicesC
   override def put(correlationId: UUID, eligibility: EligibilityResponse, nino: String)(
     implicit ec: ExecutionContext
   ): Future[Either[String, Unit]] =
-    doCreateOrUpdate(
-      correlationId.toString,
-      "eligibility",
-      Json.toJson(EligibilityResponseWithNINO(eligibility, nino))
-    ).map[Either[String, Unit]] { dbUpdate =>
+    preservingMdc {
```
Comment (johno1985) [inline app/uk/gov/hmrc/helptosaveapi/repo/EligibilityStore.scala:100]: not needed as function `doCreateOrUpdate` already has this wrapped

**Review by johno1985 — CHANGES_REQUESTED** [review]
minor change

File: app/uk/gov/hmrc/helptosaveapi/repo/EligibilityStore.scala [hunk 2]
```diff
@@ -72,21 +72,23 @@ class MongoEligibilityStore @Inject() (mongoComponent: MongoComponent, servicesC
...
-                     (data \ "eligibility")
-                       .validate[EligibilityResponseWithNINO]
-                       .asEither
-                       .leftMap(e => s"Could not parse data: ${e.mkString("; ")}")
-                   )
-        } yield result
-
-        response.value
-      }
+    preservingMdc{
```
Comment (johno1985) [inline app/uk/gov/hmrc/helptosaveapi/repo/EligibilityStore.scala:75]: this one can go also

**Review by johno1985 — CHANGES_REQUESTED** [review]
another small change, plus there is a compiler warning about unused import, might be worth doing an optimise import on the project

**Review by johno1985 — APPROVED** [review]

**Pull request closed without merging**